.github/workflows/build-and-publish.yaml

```diff
@@ -0,0 +1,53 @@
+name: Build and publish
+
+on:
+  push:
+    branches:
+      - main
+  workflow_dispatch:
+    inputs:
+      docker_tag_prefix:
+        description: "Enter the Docker Tag Prefix (e.g adhoc-test)"
+        required: true
+        type: string
+
+env:
+  ECR_REPOSITORY_URL: 673156464838.dkr.ecr.us-west-2.amazonaws.com
+  ECR_REPOSITORY_NAME: gpt-survey-summarizer
+  DOCKER_TAG_PREFIX: ${{ github.event.inputs.docker_tag_prefix }}
```

Comment:
**suggestion:** I think we should implement the use of this docker tag prefix in the workflow.

Reply:
The [implementation](https://github.com/MinaFoundation/uptime-service-backend/blob/ea827528d1a17bb75746e11ef222fd02bbe114d6/.github/workflows/publish.yaml#L34-L44) in another workflow has stood the test of time so I think it is safe to implement.